js/src/evaluate.js

```diff
@@ -3,49 +3,76 @@
  * @author Nicholas C. Zakas
  */
 
+//-----------------------------------------------------------------------------
+// Typedefs
+//-----------------------------------------------------------------------------
+
+/** @typedef {import("./typings").Node} Node */
+/** @typedef {import("./typings").NodeParts} NodeParts */
+/** @typedef {import("./typings").DocumentNode} DocumentNode */
+/** @typedef {import("./typings").StringNode} StringNode */
+/** @typedef {import("./typings").NumberNode} NumberNode */
+/** @typedef {import("./typings").BooleanNode} BooleanNode */
+/** @typedef {import("./typings").MemberNode} MemberNode */
+/** @typedef {import("./typings").ObjectNode} ObjectNode */
+/** @typedef {import("./typings").ElementNode} ElementNode */
+/** @typedef {import("./typings").ArrayNode} ArrayNode */
+/** @typedef {import("./typings").NullNode} NullNode */
+/** @typedef {import("./typings").AnyNode} AnyNode */
+/** @typedef {import("./typings").JSONValue} JSONValue */
+
 //-----------------------------------------------------------------------------
 // Exports
 //-----------------------------------------------------------------------------
 
 /**
  * Evaluates a Momoa AST node into a JavaScript value.
- * @param {Node} node The node to interpet.
- * @returns {*} The JavaScript value for the node. 
+ * @param {AnyNode} node The node to interpet.
+ * @returns {JSONValue} The JavaScript value for the node. 
  */
 export function evaluate(node) {
     switch (node.type) {
-    case "String":
-    case "Number":
-    case "Boolean":
-        return node.value;
+        case "String":
+            return node.value;
+
+        case "Number":
+            return node.value;
+
+        case "Boolean":
+            return node.value;
```

Comment:
I know this is already merged, but I bet you can keep the combined `cases` and the code should still work. The type of `node` will be `StringNode | NumberNode | BooleanNode`, but that combined type still has a `.value` property.